sdk/identity/identity/src/util/processUtils.ts

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as childProcess from "child_process";
+
+/**
+ * Easy to mock childProcess utils.
+ * @internal
+ */
+export const processUtils = {
+  /**
+   * Promisifying childProcess.execFile
+   * @internal
+   */
+  execFile(file: string, params: string[]): Promise<string | Buffer> {
+    return new Promise((resolve, reject) => {
+      childProcess.execFile(file, params, (error, stdout, stderr) => {
+        if (error || stderr) {
```

Comment:
minor quibble: sometimes binaries log to stderr without exiting unsuccessfully, but I think this is good enough for now unless Azure Powershell does this behavior regularly

Reply:
Based on our understanding of the output we expect in all of the cases we are aware of, we don't expect PowerShell to log good responses to stderr.

Reply:
Actually ran into something similar with Docker, which spits out information output to stderr in certain scenarios (even though the command succeeds). Curious if we know for sure we need to check stderr - is there a specific scenario you are thinking of or ran into? If not, maybe just checking error is sufficient?

Reply:
I think I'd rather catch stderr than ignore it 🤔 I'm more concerned on letting the good cases pass and any potential bad case to bubble up.